examples/core-auth-nextjs-email/src/app/auth/signin/actions.ts

```diff
@@ -0,0 +1,48 @@
+'use server';
+import { AuthError } from 'next-auth';
+import type { AuthProvider } from '@toolpad/core';
+import { signIn as signInAction } from '../../../auth';
+
+async function signIn(provider: AuthProvider, formData: FormData, callbackUrl?: string) {
+  try {
+    return await signInAction(provider.id, {
+      ...(formData && { email: formData.get('email'), password: formData.get('password') }),
+      redirectTo: callbackUrl ?? '/',
+    });
+  } catch (error) {
+    // The desired flow for successful sign in in all cases
+    // and unsuccessful sign in for OAuth providers will cause a `redirect`,
+    // and `redirect` is a throwing function, so we need to re-throw
+    // to allow the redirect to happen
+    // Source: https://github.com/vercel/next.js/issues/49298#issuecomment-1542055642
+    // Detect a `NEXT_REDIRECT` error and re-throw it
+    if (error instanceof Error && error.message === 'NEXT_REDIRECT') {
+      // For the nodemailer provider, we want to return a success message
+      // if the error is a 'verify-request' error, instead of redirecting
+      // to a `verify-request` page
+      if (provider.id === 'nodemailer' && (error as any).digest?.includes('verify-request')) {
```

Comment:
I guess they don't provide types that we can use to type this error with the `digest`?

Reply:
I agree that this is quite fragile, but I couldn't find any way to detect exclusively that the redirect is to `verify-request` using an Error type. I'll investigate a bit more

Reply:
Ok, no big deal if you can't find it, was just wondering.